pkg/crc/api/websocket/server.go

```diff
@@ -0,0 +1,126 @@
+package websocket
+
+import (
+	goContext "context"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/crc-org/crc/pkg/crc/logging"
+	"github.com/crc-org/crc/pkg/crc/machine"
+	"nhooyr.io/websocket"
+)
+
+type wsClient struct {
+	msgs      chan []byte
+	closeSlow func()
+}
+
+type Server struct {
+	// clientMessageBuffer controls the max number
+	// of messages that can be queued for a subscriber
+	// before it is kicked.
+	//
+	// Defaults to 16.
+	clientMessageBuffer int
+
+	// serveMux routes the various endpoints to the appropriate handler.
+	serveMux http.ServeMux
+
+	endpoints map[string]Endpoint
+}
+
+func NewWebsocketServer(machine machine.Client) *Server {
+	ws := &Server{
+		clientMessageBuffer: 16,
+		endpoints:           map[string]Endpoint{},
+	}
+
+	ws.serveMux.HandleFunc("/status", func(writer http.ResponseWriter, request *http.Request) {
+		ws.endpoints["/status"] = createStatusEndpoint(machine)
```

Comment:
Shouldn't this be done outside of `HandleFunc`? Won't this create a new status endpoint each time we get a request on `/socket/status`?